frame/support/src/signed_extensions.rs

```diff
@@ -0,0 +1,327 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Signed Extensions utilities for FRAME.
+
+use sp_runtime::{
+	codec::{Decode, Encode},
+	traits::{DispatchInfoOf, PostDispatchInfoOf, SignedExtension},
+	transaction_validity::{TransactionPriority, TransactionValidity, TransactionValidityError},
+	DispatchResult,
+};
+use sp_std::{fmt::Debug, marker::PhantomData, vec::Vec};
+
+/// Adjust transaction `priority` returned by the signed extension.
+///
+/// The idea for this type is to be able to compose multiple signed extensions
+/// and adjust the priority they are returning so that the summed up priority
+/// makes sense for a particular runtime.
+///
+/// Example:
+/// We combine [`frame_system::CheckWeight`] extension and
+/// [`frame_transaction_payment::ChargeTransactionPayment`] extensions.
+///
+/// Each of them add to `priority`, but the weight is much less important
```

Comment:
Maybe we should add some docs above each extension on what they, e.g. `priority`.

Reply:
Added docs to major `SignedExtension`s.